infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -457,13 +471,12 @@ private synchronized void detectorTaskScheduler() {
      * @return True if node is allowed to handle reconfigurations. False otherwise.
      */
     private boolean canHandleReconfigurations() {
-
         // We check for the following condition here: If the node is NOT a part of the
         // current layout, it should not attempt to change layout.
+
         Layout layout = serverContext.getManagementLayout();
-        if (!layout.getAllServers().contains(getLocalEndpoint())) {
-            log.debug("This Server is not a part of the active layout. "
-                    + "Aborting reconfiguration handling.");
+
+        if (!layout.getAllServersNodes().contains(getLocalEndpoint())) {
             return false;
```

Comment:
I would flip the logic.. 
******
if(layout.getAllServerNodes().contains(getLocakEndpoint())
   return true;

return false;
******

If this is a cluster of 3 and if cluster is stable then you can be more efficient by checking for a state that is more prevalent.

Reply:
fixed